### PR TITLE
Added Tests of stressful conditions

### PR DESCRIPTION
**Pull request by EtgarSH**
1. A suite test that check stressful conditions on the bridge.
2. The RN test project is now maintainable for lot of tests (renderTestButton).

### CODE REVIEW

File: detox/test/e2e/c-stressful-use.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+const MULTI_TEST_COUNT = 35
+
+describe.only('Stressful Use', function() {
```
Comment (talkol) [inline detox/test/e2e/c-stressful-use.js]: don't push describe.only to master.. it removes all other tests

File: detox/test/e2e/c-stressful-use.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  describe('Stress on React Bridge', function() {
+    beforeEach(function (done) {
+      simulator.reloadReactNativeApp(done)
+    })
+
+    it('Should say hello after tap in conditions of Stressful Bridge', function() {
```
Comment (talkol) [inline detox/test/e2e/c-stressful-use.js]: 1. naming convention for "Should" is "should"

2. change to "should handle click during conditions of stressful bridge"

File: detox/test/e2e/c-stressful-use.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+const MULTI_TEST_COUNT = 35
+
+describe.only('Stressful Use', function() {
+  describe('Stress on React Bridge', function() {
```
Comment (talkol) [inline detox/test/e2e/c-stressful-use.js]: we don't need so many `describe` blocks, have just one `describe` on top and put all the tests as `it` inside this describe, we don't need a describe per test

File: detox/test/e2e/c-stressful-use.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      expect(element(by.label('Hello World!!!'))).toBeVisible()
+    })
+  })
+
+  describe('Stress on JS Event Loop', function() {
+    beforeEach(function (done) {
+      simulator.reloadReactNativeApp(done)
+    })
+
+    it('Should say hello after tap in conditions of Stressful JS Events Loop', function() {
```
Comment (talkol) [inline detox/test/e2e/c-stressful-use.js]: change to "should handle tap during conditions of stressful JS event loop"

File: detox/test/e2e/c-stressful-use.js (hunk 1 above)
Comment (EtgarSH) [inline detox/test/e2e/c-stressful-use.js]: OOPS...

File: detox/test/e2e/c-stressful-use.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+      element(by.label('Events Stress')).tap()
+      expect(element(by.label('Hello World!!!'))).toBeVisible()
+    })
+  })
+
+  describe('Multi Simple Flow Test', function() {
+    for (let i = 0; i < MULTI_TEST_COUNT; i++) {
+      describe(`#${i}`, function() {
```
Comment (talkol) [inline detox/test/e2e/c-stressful-use.js]: you can remove both `describes`, the loop here should be over `it`

File: detox/test/e2e/c-stressful-use.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        beforeEach(function (done) {
+          simulator.reloadReactNativeApp(done)
+        })
+
+        it('should show hello screen after tap', function () {
```
Comment (talkol) [inline detox/test/e2e/c-stressful-use.js]: change to: "should handle tap in stress consecutive test #${i}"

File: detox/test/index.ios.js [hunk 7]
```diff
@@ -10,46 +10,97 @@ import {
   StyleSheet,
   Text,
   View,
+  ScrollView,
   TouchableOpacity
 } from 'react-native';
+import _ from 'lodash';
+
+const STRESSFUL_STRING_LENGTH = 7800
```
Comment (talkol) [inline detox/test/index.ios.js]: our conventions is to have ; in the end of all lines, please fix in all the code

File: detox/test/index.ios.js [hunk 8]
```diff
@@ -10,46 +10,97 @@ import {
...
       </View>
     );
   }
   renderAfterButton() {
     return (
       <View style={{flex: 1, paddingTop: 20, justifyContent: 'center', alignItems: 'center'}}>
-        <Text style={{fontSize: 25}}>
-          {this.state.greeting}!!!
-        </Text>
+        <ScrollView>
```
Comment (talkol) [inline detox/test/index.ios.js]: why do we need a scrollview here?

File: detox/test/index.ios.js [hunk 9]
```diff
@@ -10,46 +10,97 @@ import {
...
   }
+
   onButtonPress(greeting) {
     this.setState({
       greeting: greeting
     });
   }
+
+  bridgeStressButtonPressed(greeting) {
+    this.onButtonPress(greeting)
```
Comment (talkol) [inline detox/test/index.ios.js]: don't have 2 calls to `setState`, combine them to one call together, no need to call `onButtonPress`, just do the combined `setState` here. I want to make sure we send both things together over the bridge

File: detox/test/index.ios.js [hunk 10]
```diff
@@ -10,46 +10,97 @@ import {
...
+    const data = buildStringByLength(STRESSFUL_STRING_LENGTH)
+    this.setState({
+      passToBridge: data
+    })
+  }
+
+  eventsStressButtonPressed(greeting) {
+    // Stress:
+    for (let i =0; i < STRESSFUL_EVENTS_COUNT; i++) {
+      let myString = ""
```
Comment (talkol) [inline detox/test/index.ios.js]: no need to define myString outside the block, move it to the block completely

File: detox/test/index.ios.js [hunk 11]
```diff
@@ -10,46 +10,97 @@ import {
...
+      setImmediate(() => { myString = buildStringByLength(1000) })
+    }
+
+    this.onButtonPress(greeting)
```
Comment (talkol) [inline detox/test/index.ios.js]: this must run in `setImmediate` too or else you don't wait in line for all the events to be processed

**Pull request merged**